test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -422,13 +435,20 @@ struct usm_vector
     {
         assert(vec.size() == n);
     }
+    template<typename Policy, typename DataGen>
+    usm_vector(Policy&& exec, int n, DataGen gen): vec(n, shared_allocator(exec.queue()))
+    {
+        for(int i = 0; i < n; ++i)
+            vec[i] = gen(i);
+    }
     type& operator()()
     {
         return vec;
     }
     ~usm_vector()
     {
-        std::copy_n(vec.begin(), vec.size(), p);
+        if(p)
```

Comment:
The type has still another constructor which allows to create an instance by C style host array (pointer and size):
`usm_vector(Policy&& exec, T* data, int n)`. It can be useful.

Reply:
Ok. I do not mind leaving it as is.